packages/backend/src/handlers/triggers/updateDiscordRole.ts

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable no-console */
+import Discord, { Role } from 'discord.js';
+
+import { CONFIG } from '../../config';
+import { AccountType_Enum, Player, PlayerRank_Enum } from '../../lib/autogen/hasura-sdk';
+import { client } from '../../lib/hasuraClient';
+import { TriggerPayload } from './types';
+
+export interface UpdateRole {
+  playerId: string;
+  previousRole: string | undefined;
+  newRole: string;
+}
+
+export const updateDiscordRole = async (
+  payload: TriggerPayload<Player>,
+) => {
+  const {old: oldPlayer, new: newPlayer} = payload.event.data;
+
+  try {
+    if (newPlayer == null) return;
+   
+    const getPlayerResponse = await client.GetPlayer({playerId: newPlayer.id});
+    const discordPlayerAccount = getPlayerResponse.player_by_pk?.Accounts?.find(a => a.type === AccountType_Enum.Discord);
+    if (discordPlayerAccount?.identifier == null) return;
+
+    const newRank = newPlayer?.rank;
+
+    if (newRank == null) return;
+   
+    // look up guild by guildname = 'metagame' (for now), 
+    const getGuildResponse = await client.GetGuild({guildname: 'metafam'});
+    const discordAccount = getGuildResponse.guild[0]?.guild_accounts?.find(a => a.type === AccountType_Enum.Discord);
+    if (discordAccount == null) return;
+
+    // call discord API. We'll need serverId, playerId, and roleIds
+    const discordClient = new Discord.Client();
+    discordClient.login(CONFIG.discordBotToken);
+
+    // todo add jsonb field to guild and populate ranks, e.g. { ranks: [] }
+    const guild = await discordClient.guilds.fetch(discordAccount.identifier);
+    if (guild == null) {
```

Comment:
Why not just
```suggestion
    if (!guild) {
```

Reply:
I prefer `==` null as it is more explicit and typesafe. `!false` is true, but `false == null` is still false.

Reply:
* `undefined == null` → `true`
* `undefined === null` → `false`

It's safest to use `===` as often as possible.

Reply:
>     * `undefined == null` → `true`
> 
>     * `undefined === null` → `false`
> 
> 
> It's safest to use `===` as often as possible.

Yes, so if `guild` is `undefined` or `null`, return. In my experience, one usually doesn't need to worry about the distinction.